### DISCUSSION TITLE
Cell内の値の受け渡し

### DISCUSSION THREAD

**Discussion by mahirooo** [lit-iphone-course/lit-iphone-course#36] [Question-original] [answered]
(1) 実現したいこと
ViewController内の、CollectionView内のCellの中の値３つ(todoLabelとrewardLabelとbackgroundImageView.image)を遷移先であるPushViewControllerに受け渡したい
(やりたいことは「選択したCellの内容をそのまま遷移先に表示させたい」ので、Cellの内容をそのまま受け渡せるならそれができると嬉しいです！)

(2) 実現するために自分が試した内容とその結果
❶https://qiita.com/fromage-blanc/items/4ab850f121824310275bを参考にコードを書いてみたが、うまく受け渡せなかった(エラーなし)
❷override func prepareを使用して、遷移先に用意した変数に「cell.todoLabel.text」と「cell.rewardLabel.text」を入れようとした。
→エラー文「Cannot find 'cell' in scope」

(3) 調べた内容や資料
https://qiita.com/fromage-blanc/items/4ab850f121824310275b
https://softmoco.com/basics/how-to-implement-table-view-navigation.php#2-1

(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
**Viewcontroller内**

```
internal func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
     
            // セルの選択を解除
        collectionView.cellForItem(at: indexPath)
     
            // 別の画面に遷移
            performSegue(withIdentifier: "toNextViewController", sender: saveItem)
        print("選択しました: \(indexPath.row)")
        }
```
[gitの行リンク](https://github.com/mahirooo/TampStamp/blob/ef4cde10f09abaeb0980f92028c8a72a6076b47d/TampStamp/ViewController.swift#L82-L90)

```
override func prepare(for segue: UIStoryboardSegue, sender: Any?) {

            let nextVC = segue.destination as! PushViewController
        let _ = nextVC.view
        nextVC.todo = cell.todoLabel.text
        nextVC.reward = cell.rewardLabel.text
        }
```
[gitの行リンク](https://github.com/mahirooo/TampStamp/blob/ef4cde10f09abaeb0980f92028c8a72a6076b47d/TampStamp/ViewController.swift#L101-L107)

(5) GitHubのリンク
https://github.com/mahirooo/TampStamp

**Comment by txxxxc** [comment, marked as answer]
cellはprepare(for:sender:)からアクセスすることが出来ないので、cellに反映しているデータ(saveItemの中の一つ)をprepare関数に渡してあげましょう！

具体的に言うと
1. `func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath)`の中で使えるindexPathを使ってsaveItem内の一つの要素をprepareに渡す
2. senderから渡した値を使ってnextVCに値を渡す
みたいな流れでやるとうまくいくと思います！
具体的にはこんな感じです！！
```swift
    internal func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
     
            // セルの選択を解除
        collectionView.cellForItem(at: indexPath)
     
            // 別の画面に遷移
        performSegue(withIdentifier: "toNextViewController", sender: saveItem[indexPath.row])
        print("選択しました: \(indexPath.row)")
        }

```
```swift
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {

        let nextVC = segue.destination as! PushViewController
        let item = sender as! Save
        nextVC.todo = item.todo
        nextVC.reward = item.reward
    }
```

**Reply by mahirooo** [reply]
アクセスできないんですね....！(知らなかった)上記記述したら行けました！！！ありがとうございます！！！
